arborist/token.go

```diff
@@ -46,38 +46,39 @@ func (server *Server) decodeToken(token string, scopes []string) (*TokenInfo, er
 	if !casted {
 		return nil, fieldTypeError("context")
 	}
-	userInterface, exists := context["user"]
-	if !exists {
-		return nil, missingRequiredField("user")
-	}
-	user, casted := userInterface.(map[string]interface{})
-	if !casted {
-		return nil, fieldTypeError("user")
-	}
-	usernameInterface, exists := user["name"]
-	if !exists {
-		return nil, missingRequiredField("name")
-	}
-	username, casted := usernameInterface.(string)
-	if !casted {
-		return nil, fieldTypeError("name")
-	}
-	policiesInterface, exists := user["policies"]
+	username := ""
 	var policies []string = nil
-	// it's ok if there's no policies in the token; we'll just look up the username
+	userInterface, exists := context["user"]
+	// it's ok if there's no user; it's a client credentials token
 	if exists {
-		// policiesInterface should really be a []string, so cast all the elements
-		policiesInterfaceSlice, casted := policiesInterface.([]interface{})
+		user, casted := userInterface.(map[string]interface{})
 		if !casted {
-			return nil, fieldTypeError("policies")
+			return nil, fieldTypeError("user")
+		}
+		usernameInterface, exists := user["name"]
+		if !exists {
+			return nil, missingRequiredField("name")
 		}
-		policies := make([]string, len(policiesInterfaceSlice))
-		for i, policyInterface := range policiesInterfaceSlice {
-			policyString, casted := policyInterface.(string)
+		username, casted = usernameInterface.(string)
+		if !casted {
+			return nil, fieldTypeError("name")
```

Comment:
should be username here i think

Reply:
we call the variable "username" but in the token it's "name" - see `usernameInterface, exists := user["name"]`

Reply:
i just indented this block, it was already checking for `name` before

i think it might be a bit clearer to throw the error with `context.user.name` instead of just `name`, but we'd need to go through the rest of the code to make sure the errors are consistent and i'd rather avoid the scope creep in this PR since it's already a big feature